### ISSUE TITLE
Caching the parent can cause issues with drawing the section box

### ISSUE THREAD

**Issue by githubdorey** [Fuzzyma/svg.select.js#23]
If the svg element was selected and then you move this element to a new parent.  Once moved, then remove or hide the original parent, the selection box will not be drawn or will be hidden because it still wants to draw it in the original parent.

**Comment by Fuzzyma** [comment]
Only thing we can do here to move the parent-caching in the init method. So any call to selectice on the element will actualize it - would this fix it?

**Comment by githubdorey** [comment]
I did just that and after a day of testing looks to fix the issue and not seeing any side affects.
```
SelectHandler.prototype.init = function (value, options) {

        var bbox = this.el.bbox();
        this.options = {};

        // Merging the defaults and the options-object together
        for (var i in this.el.selectize.defaults) {
            this.options[i] = this.el.selectize.defaults[i];
            if (options[i] !== undefined) {
                this.options[i] = options[i];
            }
        }

        this.parent = this.el.parent(); // adding this fix the problem for me
        this.nested = (this.nested || this.parent.group());
        this.nested.matrix(new SVG.Matrix(this.el).translate(bbox.x, bbox.y));

        // When deepSelect is enabled and the element is a line/polyline/polygon, draw only points for moving
        if (this.options.deepSelect && ['line', 'polyline', 'polygon'].indexOf(this.el.type) !== -1) {
            this.selectPoints(value);
        } else {
            this.selectRect(value);
        }

        this.observe();
        this.cleanup();

    };
```

**Comment by Fuzzyma** [comment]
Fixed and tagged as 2.1.1

**Comment by cmatzenbach** [comment]
This is causing me problems.  When I import an svg element per the instructions [here](svgjs.com/importing-exporting/#import-export-svg) and try to add a click handler that calls selectize, it is passing in the create object instead of the initializer, and I get the error "this.parent.group is undefined".   Maybe this is something I am misunderstanding about the library?
